lib/java/src/org/apache/thrift/transport/TFramedTransport.java

```diff
@@ -123,6 +123,12 @@ public void consumeBuffer(int len) {
     readBuffer_.consumeBuffer(len);
   }
 
+  public void clear() {
+    if (readBuffer_ != null) {
```

Comment:
`readBuffer_` is declared `private TMemoryInputTransport readBuffer_ = new TMemoryInputTransport(new byte[0]);` and never re-assigned in this class. Unless there is known code re-assigning via reflection, this should just be promoted to final and all the null checks - like the one added here - dispensed with.